pkg/actors/actors.go

```diff
@@ -437,96 +437,115 @@ func (a *actorsRuntime) constructActorStateKey(actorType, actorID, key string) s
+	stream := a.newPlacementStreamConn(placementAddress)
 
+	// Establish receive channel to retrieve placement table update
 	go func() {
 		for {
-			host := placementv1pb.Host{
-				Name:     hostAddress,
-				Load:     1,
-				Entities: a.config.HostedActorTypes,
-				Port:     int64(a.config.Port),
-				Id:       a.config.AppID,
+			// Wait until stream is reconnected.
+			if !isConnAlive || stream == nil {
+				time.Sleep(placementReconnectInterval)
+				continue
 			}
 
-			if stream != nil {
-				if !a.appHealthy {
-					// app is unresponsive, close the stream and disconnect from the placement service
-					err := stream.CloseSend()
-					if err != nil {
-						log.Errorf("error closing stream to placement service: %s", err)
-					}
-					continue
-				}
+			resp, err := stream.Recv()
+			if err != nil {
+				diag.DefaultMonitoring.ActorStatusReportFailed("recv", "status")
+				log.Warnf("failed to receive placement table update from placement service: %v", err)
 
-				if err := stream.Send(&host); err != nil {
-					diag.DefaultMonitoring.ActorStatusReportFailed("send", "status")
-					log.Warnf("failed to report status to placement service : %v", err)
-					stream = a.getPlacementClientPersistently(placementAddress, hostAddress)
-				}
+				// If receive channel is down, ensure the current stream is closed and get new gRPC stream.
+				stream.CloseSend()
+				isConnAlive = false
+				stream = a.newPlacementStreamConn(placementAddress)
+				isConnAlive = true
+
+				continue
 			}
-			time.Sleep(heartbeatInterval)
+
+			diag.DefaultMonitoring.ActorStatusReported("recv")
+			a.onPlacementOrder(resp)
 		}
 	}()
 
+	// Send the current host status to placement to register the member and
+	// maintain the status of member by placement.
 	go func() {
 		for {
-			resp, err := stream.Recv()
-			if err != nil {
-				diag.DefaultMonitoring.ActorStatusReportFailed("recv", "status")
-				log.Warnf("failed to receive the response of status report from placement service: %v", err)
-				stream = a.getPlacementClientPersistently(placementAddress, hostAddress)
-			} else {
-				diag.DefaultMonitoring.ActorStatusReported("recv")
+			// Wait until stream is reconnected.
+			if !isConnAlive || stream == nil {
+				time.Sleep(placementReconnectInterval)
+				continue
 			}
-			if resp != nil {
-				a.onPlacementOrder(resp)
+
+			// appHealthy is the health status of actor service application. This allows placement to update
+			// memberlist and hashing table quickly.
+			if !a.appHealthy {
+				// app is unresponsive, close the stream and disconnect from the placement service.
+				// Then Placement will remove this host from the member list.
+				err := stream.CloseSend()
+				if err != nil {
+					log.Errorf("error closing stream to placement service: %s", err)
+				}
+				continue
+			}
+
+			host := placementv1pb.Host{
+				Name:     fmt.Sprintf("%s:%d", hostAddress, a.config.Port),
+				Load:     1, // TODO: Use CPU/Memory as load balancing factor
```

Comment:
When you say CPU/Memory, do you mean realtime available CPU/Memory, or simply static host size? 

CPU/Memory tends to miss other performance problems such as disk and networking. Things like the ratio of busy time/idle time can capture system load more reliably. Or work queue depth, if your application has something like that.

Reply:
We will not support dynamic load balancing before 1.0. No plan but I would like to add comment, how we can use this number for the future. Disk/network can definitely be considered as load factor.

Reply:
Well, specifically in an event driven system, a good measure that can capture slowdowns of any kind is (time_processing_events)/(time_processing_events + time_waiting_for_new_events). This manages to capture the aggregate of CPU + Network + disk + memory + any other thing that might slow down event processing.